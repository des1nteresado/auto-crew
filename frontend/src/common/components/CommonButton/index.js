import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';

const CommonButton = ({ label, isDisabled, ...otherProps }) => {
  return (
    <Button disabled={isDisabled} disableRipple disableFocusRipple {...otherProps}>
      {label}
    </Button>
  );
};

CommonButton.defaultProps = {
  label: 'Button',
  isDisabled: false,
};

CommonButton.propTypes = {
  isDisabled: PropTypes.bool,
  label: PropTypes.string,
};

export default React.memo(CommonButton);
