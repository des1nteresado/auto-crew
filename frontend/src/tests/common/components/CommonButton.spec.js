import React from 'react';
import { mount } from 'enzyme';

import CommonButton from 'common/components/CommonButton';

const defaultProps = {
  onClick: jest.fn(),
};

const setUp = (props) => mount(<CommonButton {...defaultProps} {...props} />);

describe('CommonButton', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders CommonButton component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls correct handler on button click', () => {
    const button = component.find('button');
    button.simulate('click');
    expect(defaultProps.onClick).toHaveBeenCalled();
  });
});
