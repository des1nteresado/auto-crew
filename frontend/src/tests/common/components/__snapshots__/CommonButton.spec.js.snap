// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonButton renders CommonButton component correctly 1`] = `
<CommonButton
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(Button))
    disableFocusRipple={true}
    disableRipple={true}
    disabled={false}
    onClick={[MockFunction]}
  >
    <ForwardRef(Button)
      classes={
        Object {
          "colorInherit": "MuiButton-colorInherit",
          "contained": "MuiButton-contained",
          "containedPrimary": "MuiButton-containedPrimary",
          "containedSecondary": "MuiButton-containedSecondary",
          "containedSizeLarge": "MuiButton-containedSizeLarge",
          "containedSizeSmall": "MuiButton-containedSizeSmall",
          "disableElevation": "MuiButton-disableElevation",
          "disabled": "Mui-disabled",
          "endIcon": "MuiButton-endIcon",
          "focusVisible": "Mui-focusVisible",
          "fullWidth": "MuiButton-fullWidth",
          "iconSizeLarge": "MuiButton-iconSizeLarge",
          "iconSizeMedium": "MuiButton-iconSizeMedium",
          "iconSizeSmall": "MuiButton-iconSizeSmall",
          "label": "MuiButton-label",
          "outlined": "MuiButton-outlined",
          "outlinedPrimary": "MuiButton-outlinedPrimary",
          "outlinedSecondary": "MuiButton-outlinedSecondary",
          "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
          "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
          "root": "MuiButton-root",
          "sizeLarge": "MuiButton-sizeLarge",
          "sizeSmall": "MuiButton-sizeSmall",
          "startIcon": "MuiButton-startIcon",
          "text": "MuiButton-text",
          "textPrimary": "MuiButton-textPrimary",
          "textSecondary": "MuiButton-textSecondary",
          "textSizeLarge": "MuiButton-textSizeLarge",
          "textSizeSmall": "MuiButton-textSizeSmall",
        }
      }
      disableFocusRipple={true}
      disableRipple={true}
      disabled={false}
      onClick={[MockFunction]}
    >
      <WithStyles(ForwardRef(ButtonBase))
        className="MuiButton-root MuiButton-text"
        component="button"
        disableRipple={true}
        disabled={false}
        focusRipple={false}
        focusVisibleClassName="Mui-focusVisible"
        onClick={[MockFunction]}
        type="button"
      >
        <ForwardRef(ButtonBase)
          className="MuiButton-root MuiButton-text"
          classes={
            Object {
              "disabled": "Mui-disabled",
              "focusVisible": "Mui-focusVisible",
              "root": "MuiButtonBase-root",
            }
          }
          component="button"
          disableRipple={true}
          disabled={false}
          focusRipple={false}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[MockFunction]}
          type="button"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[MockFunction]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Button
            </span>
          </button>
        </ForwardRef(ButtonBase)>
      </WithStyles(ForwardRef(ButtonBase))>
    </ForwardRef(Button)>
  </WithStyles(ForwardRef(Button))>
</CommonButton>
`;
